modules/generator/generator_test.go

```diff
@@ -129,3 +137,140 @@ func (l testLogger) Log(keyvals ...interface{}) error {
 	l.t.Log(keyvals...)
 	return nil
 }
+
+func BenchmarkPushSpans(b *testing.B) {
```

Comment:
I love the full stack benchmark.  This is something we might want to do for other components as well.